deployment/tfhub/object-detection-base64.ipynb

```diff
@@ -0,0 +1,627 @@
+        "    def __init__(self, artifacts=None):\n",
+        "        module_handle = 'https://tfhub.dev/google/openimages_v4/ssd/mobilenet_v2/1'\n",
+        "        self.detector = hub.load(module_handle).signatures['default']\n",
+        "    \n",
+        "    def handle_img(self, img, width=640, height=480):\n",
+        "        _, path = tempfile.mkstemp(suffix='.jpg')\n",
+        "        img_str = json.loads(img)\n",
+        "        img_bytes = img_str.encode('utf-8')\n",
+        "        img_bytes = io.BytesIO(base64.b64decode(img_bytes))\n",
+        "        img_arr = np.array(Image.open(img_bytes), dtype=np.uint8)\n",
+        "        img = Image.fromarray(img_arr)\n",
+        "        img = ImageOps.grayscale(img)\n",
+        "        img.thumbnail((width, height), Image.LANCZOS)\n",
+        "        img.save(path)\n",
+        "        \n",
+        "        print(f\"Image downloaded to {path}.\")\n",
+        "        return path\n",
+        "\n",
+        "    def load_img(self, path):\n",
+        "        img = tf.io.read_file(path)\n",
+        "        img = tf.image.decode_jpeg(img, channels=3)\n",
+        "        \n",
+        "        return img\n",
+        "\n",
+        "    def filter_results(self, file, response, entity='Car', min_score=.2):\n",
+        "        unused_keys = ['detection_class_labels', 'detection_class_names']\n",
+        "        response = {key: value.numpy().tolist() for key, value in response.items()}\n",
+        "        response = {key: val for key, val in response.items() if key not in unused_keys}\n",
+        "        response['detection_class_entities'] = [v.decode() for v in response['detection_class_entities']]\n",
+        "\n",
+        "        entities = response['detection_class_entities']\n",
+        "        scores = response['detection_scores']\n",
+        "        bboxes = response['detection_boxes']\n",
+        "        result = {}\n",
+        "\n",
+        "        for i in range(len(entities)):\n",
+        "            if entities[i] == entity and scores[i] >= min_score:\n",
+        "                ymin, xmin, ymax, xmax = bboxes[i]\n",
+        "                result = {\n",
+        "                    'file': file,\n",
+        "                    'has_car': True,\n",
+        "                    'score': scores[i],\n",
+        "                    'bboxes': {'ymin': ymin, 'xmin': xmin, 'ymax': ymax, 'xmax': xmax}\n",
+        "                }\n",
+        "                break\n",
+        "\n",
+        "        if len(result) == 0:\n",
+        "            result = {\n",
+        "                'file': file,\n",
+        "                'has_car': False,\n",
+        "                'score': 0,\n",
+        "                'bboxes': {'ymin': 0, 'xmin': 0, 'ymax': 0, 'xmax': 0}\n",
+        "            }\n",
+        "        \n",
+        "        print(f\"Found {result['has_car']} object(s).\")\n",
```

Comment:
Since we changed `has_car` from `int` to `bool`, this message should probably be changed (or removed).

Reply:
Done.